### PR TITLE
Add validation to role_id

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

**Review by zied-elouaer — DISMISSED** [review]
LGTM

**Comment by soerenmartius** [comment]
@mariux will wait for you to give it a second check :)

File: variables.tf [hunk 1]
```diff
@@ -6,6 +6,11 @@
 variable "role_id" {
   type        = string
   description = "(Required) The camel case role id to use for this role. Cannot contain - characters."
+
+  validation {
+    condition     = can(regex("^[a-z].*[a-zA-Z0-9_.-]*$", var.role_id))
```
Comment (mariux) [inline variables.tf:11]: the condition is not matching the required format ;)

**Review by mariux — DISMISSED** [review]

File: variables.tf (hunk 1 above)
Comment (soerenmartius) [reply variables.tf:11]: please add a suggestion :)

Comment (waxb) [inline variables.tf:11]:
```suggestion
    condition     = can(regex("^[a-z](([a-z0-9]+[A-Z]?)*)$", var.role_id))
```

File: variables.tf [hunk 2]
```diff
@@ -6,6 +6,11 @@
 variable "role_id" {
   type        = string
   description = "(Required) The camel case role id to use for this role. Cannot contain - characters."
```
Comment (waxb) [inline variables.tf:8]:
```suggestion
  description = "(Required) The camel case role id to use for this role. Only can contain alphanumeric characters."
```

**Review by waxb — DISMISSED** [review]

File: variables.tf (hunk 1 above)
Comment (soerenmartius) [reply variables.tf]: resolved

File: variables.tf [hunk 3]
```diff
@@ -5,7 +5,12 @@
 
 variable "role_id" {
   type        = string
-  description = "(Required) The camel case role id to use for this role. Cannot contain - characters."
+  description = "(Required) The camel case role id to use for this role. Can only contain alphanumeric characters."
+
+  validation {
+    condition     = can(regex("^[a-z](([a-z0-9]+[A-Z]?)*)$", var.role_id))
+    error_message = "The id of the role can only contain characters and numbers, must start with a character and be defined in camel case."
```
Comment (waxb) [inline variables.tf:12]:
```suggestion
    error_message = "The id of the role can only contain alphanumeric characters, must start with a lowercase letter or number and be defined in camel case."
```

**Pull request merged**